reactioncommerce/reaction
# Fix 3945: update shopId the right way.

Resolves #3945 
Impact: **critical**  
Type: **bugfix**

## Issue
The `activeShopId` is cached on the client side [here](https://github.com/reactioncommerce/reaction/blob/master/client/modules/core/main.js#L27). So when `Rection.setUserPreferences("reaction", "activeShopId", newShopId)` is used directly the value is not cached.

## Solution
Replaced that call with `Reaction.setShopId(newShopId)` which takes care of the caching.

## Testing
1. Invite a user to marketplace.
1. Sign in as that user.
1. Observe you get `Reaction1` in the NavBar
1. Log out
1. Check localStorage to see that the `reaction = { activeShopId: "<primaryshopid>" }` is set to the primary shop.
1. Login again.
1. Observe you get `Reaction1` in the NavBar and the value is updated in the localStorage.
